### PR TITLE
My/pay as bid

### PR DESCRIPTION
**Pull request by m-yahya**
This will add CLI commands to call the Rust CLI.

### CODE REVIEW

**Comment by j-ti** [comment]
Solves https://github.com/olisystems/BEST-Energy/issues/40

File: src/lib.rs [hunk 1]
```diff
@@ -51,19 +81,82 @@ fn new_trusted_account_cmd() -> Commands {
 
 // just a skeleton to see if the cli understands it. It will never return a successful result.
 // We need to fill in some actual meaningful values.
```
Comment (j-ti) [inline src/lib.rs]: comment above not valid anymore

File: src/lib.rs [hunk 2]
```diff
@@ -3,27 +3,56 @@ use pyo3::wrap_pyfunction;
...
 
+#[pyfunction]
+fn run_cli(
+    node_url: String,
+    node_port: String,
+    worker_url: String,
+    trusted_worker_port: String,
+    command_name: String,
+    params: Vec<String>,
+) -> PyResult<()> {
```
Comment (j-ti) [inline src/lib.rs:39]: Should return value e.g. the created `account` string (which could be put also inside a dictionary) or the matches as Python Dict/List:
Could be similar to the following example for returning a python List of dictionaries:
```
    // Convert MarketOutput to Python List of Dict representing matches
    let pyMatches = PyList::new(py,
        matches.matches.into_iter().map(|mtch| {
            let dict = PyDict::new(py);
            dict.set_item("time", mtch.energy_kwh);
            dict.set_item("bid_id", mtch.bid_id);
            dict.set_item("ask_id", mtch.ask_id);
            dict.set_item("bid_actor", "");
            dict.set_item("ask_actor", "");
            dict.set_item("bid_cluster", 0);
            dict.set_item("ask_cluster", 0);
            dict.set_item("energy", mtch.energy_kwh);
            dict.set_item("price", mtch.price_euro_per_kwh);
            dict.set_item("included_grid_fee", 0);
            dict
        })
    );
```
As both types are different the account string or the match list could be it self returned within a dictionary with a specific key, i.e. {"account": account_string} or {"matches": pyMatches}

File: .env.example [hunk 3]
```diff
@@ -0,0 +1,4 @@
+NODE_URL=
```
Comment (j-ti) [inline .env.example]: default values or comment regarding formatting

File: README.md [hunk 4]
```diff
@@ -38,3 +38,77 @@ python run_integritee_cli.py
...
+```
+
+2. Open the `.env` file and provide values for the environment variables.
+3. Install the necessary dependencies by running:
+
+```bash
+$ pip install -r requirements.txt
+```
+
+## Run the Client
```
Comment (j-ti) [inline README.md]: rename "Run as Command Line Tool"

File: README.md [hunk 5]
```diff
@@ -38,3 +38,77 @@ python run_integritee_cli.py
...
+Run Rust CLI with specific commands
+
+options:
+  -h, --help            show this help message and exit
+  --command {new_account_cmd,new_trusted_account_cmd,pay_as_bid_cmd,get_market_results_cmd,pay_as_bid_proof_cmd,verify_proof_cmd}
+                        Please specify the command to run
+  --params PARAMS [PARAMS ...]
+                        Parameters for the command
+```
+
```
Comment (j-ti) [inline README.md:118]: new subsection
`### Example Commands`

File: README.md [hunk 6]
```diff
@@ -38,3 +38,77 @@ python run_integritee_cli.py
 ```
 rustup target add --toolchain nightly wasm32-unknown-unknown
 ```
+
+## Using Environment Variables
```
Comment (j-ti) [inline README.md]: insert section below within "Run as Command Line Tool"
i.e. `### Setup Environment Variables`

File: src/lib.rs [hunk 7]
```diff
@@ -36,10 +65,11 @@ fn new_account_cmd() -> Commands {
 }
 
 /// Create a new trusted account in the enclave.
-fn new_trusted_account_cmd() -> Commands {
-    Commands::Trusted(TrustedCli{
+fn new_trusted_account_cmd(params: &[String]) -> Commands {
+    let mrenclave = &params[0];
+    Commands::Trusted(TrustedCli {
         // random mrenclave, has to be replaced with one that has been fetched from the chain.
```
Comment (j-ti) [inline src/lib.rs]: update comment

File: src/lib.rs [hunk 8]
```diff
@@ -36,10 +65,11 @@ fn new_account_cmd() -> Commands {
...
-        mrenclave: "4GMb72Acyg8hnnnGEJ89jZK5zxNC4LvSe2ME96wLRV6J".to_string(),
+        mrenclave: mrenclave.to_string(),
         shard: None,
         // signers and accounts starting with `//` will be recognized as dev-seeds and can
         // always be used without first creating them in the keystore.
```
Comment (j-ti) [inline src/lib.rs:75]: Add comment explaining that //Alice is not actually the account to be created here, but that it is used as signer for the trusted call

File: src/lib.rs (hunk 2 above)
Comment (j-ti) [reply src/lib.rs:39]: with the adapted algorithm, I might add some information to the matches returned by simplyr-lib both for pay as bid as well as the cluster-based matching

**Review by j-ti — CHANGES_REQUESTED** [review]
It works
Requested changes 
- mostly update of comments
- adding return values e.g. in python dict type

File: src/lib.rs (hunk 1 above)
Comment (m-yahya) [reply src/lib.rs]: Removed.

File: .env.example (hunk 3 above)
Comment (m-yahya) [reply .env.example]: Default values added.

File: README.md (hunk 4 above)
Comment (m-yahya) [reply README.md]: Renamed.

File: README.md (hunk 5 above)
Comment (m-yahya) [reply README.md:118]: Example commands added.

File: README.md (hunk 6 above)
Comment (m-yahya) [reply README.md]: Please take a look at the updated `README` file.

File: src/lib.rs (hunk 7 above)
Comment (m-yahya) [reply src/lib.rs]: Comment updated.

File: src/lib.rs (hunk 8 above)
Comment (m-yahya) [reply src/lib.rs:75]: updated

File: src/lib.rs (hunk 2 above)
Comment (j-ti) [reply src/lib.rs:39]: Created issue https://github.com/BESTenergytrade/integritee-cli-py/issues/5 can be changed in subsequent PR

**Review by j-ti — COMMENTED** [review]
- I propose to merge https://github.com/BESTenergytrade/integritee-cli-py/pull/4 first
- https://github.com/BESTenergytrade/integritee-cli-py/pull/3/files#r1302951203  can be changed in subsequent PR

**Pull request merged**